enigma-machine-android/app/src/main/java/com/smitpatel/enigmamachine/models/EnigmaModel.kt

```diff
@@ -133,19 +104,19 @@ object EnigmaModel {
         rotorOne = Rotor.makeRotor(
             rotorOption = settings.rotorOneOption,
             position = settings.rotorOnePosition,
-            ring = settings.rotorOnePosition,
+            ring = settings.ringOneOption,
         )
 
         rotorTwo = Rotor.makeRotor(
             rotorOption = settings.rotorTwoOption,
             position = settings.rotorTwoPosition,
-            ring = settings.rotorTwoPosition,
+            ring = settings.ringTwoOption,
         )
 
         rotorThree = Rotor.makeRotor(
             rotorOption = settings.rotorThreeOption,
             position = settings.rotorThreePosition,
-            ring = settings.rotorThreePosition,
```

Comment:
This is what was causing the bug. I was setting the rotor position for the ring settings